RHEL6_7/system/pam/check

```diff
@@ -4,22 +4,57 @@
 
 #END GENERATED SECTION
 
-POSTUPGRADE_DIR="$VALUE_TMP_PREUPGRADE/postupgrade.d/pam"
+POSTUPGRADE_DIR="$POSTUPGRADE_DIR/pam"
+CLEANCONF_DIR="$VALUE_TMP_PREUPGRADE/cleanconf/etc/pam.d"
+
 if [[ ! -d "$POSTUPGRADE_DIR" ]]; then
     mkdir -p "$POSTUPGRADE_DIR"
 fi
+if [[ ! -d "$CLEANCONF_DIR" ]]; then
+    mkdir -p "$CLEANCONF_DIR"
+fi
+
 SCRIPT_NAME="postupgrade-pam.sh"
 POST_SCRIPT="postupgrade.d/$SCRIPT_NAME"
+
 cp -f $POST_SCRIPT $POSTUPGRADE_DIR/$SCRIPT_NAME
+
 fail=0
 
+comment_invalid () {
+    perl -p -i -e 's/^/#/g if /^(?!.*#)(?=.*(pam_passwdqc|pam_ecryptfs))/;' "$1"
+}
+get_deprec_dirctv () {
+
+    deprec_dirctv="$1"
+    file="$2"
+    export deprec_dirctv
+    export file 
```

Comment:
why do you export the file variable? By this you are rewritting the `$file` variable outside of the scope of this function (even when in the current call you are providing the outer `$file` content, storing it into the innter `$file` variable). Both of these exports are not needed as the whole perl script can be written without ENV at all.

Reply:
I have renamed the variable so there is no name conflict with another variable used outside the function. Thanks for spotting it! Export  as such is not a problem, and using it + $ENV{var} construct is a standard way to pass shell variable to perl.